### PR TITLE
Fix version conflicts, verify publishability of packages in CI, Remove Travis

### PR DESCRIPTION
**Pull request by gspencergoog**
This fixes a number of version resolution errors that would prevent these packages from building, and turn off Travis, since Cirrus now works.

I also fixed a number of errors that would have prevented these packages from being published without warnings, and establishes a CI script that will verify publishability for any changed packages in a PR.

For some examples, we were depending upon versions of other 1st-party plugins by using `path: ../../<package>`, which really isn't the right way to do things, so for those examples, they now depend on a published version.

I bumped the version number of any packages that were modified, and updated their CHANGELOGs since the dependencies have changed.

Also fixed a number of analyzer errors that somehow snuck in.

### CODE REVIEW

**Comment by mehmetf** [comment]
Thank you for this Greg. I started going down this path today just to fix the analyzer and build errors. I will close my PR.

**Comment by mehmetf** [comment]
It also seems like a package sneaked in (MLExample) that does not even build for iOS.

**Comment by gspencergoog** [comment]
Hmm, yes, I see what you mean.  Hey @bparrishMines, does that iOS example compile for you?  Could you take a look and give me a patch if needed? (I don't really know much about building iOS apps). I'm trying to get the plugins repo to the point that it all builds...

**Comment by gspencergoog** [comment]
I think the iOS problem is fixed now that I synced to master branch HEAD again. Now if I could get the apk build to finish before the gradle daemon crashes...

**Comment by gspencergoog** [comment]
I also had to bump the memory requirements from the default (4G?) to 8G, because the sensors example kept running out of memory and crashing during the gradle step.

File: script/check_publish.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  if [[ "${#failures[@]}" != 0 ]]; then
+    echo "FAIL: The following ${#failures[@]} package(s) failed the publishing check:"
+    for failure in "${failures[@]}"; do
+      echo "$failure"
+    done
+  fi
+  return "${#failures[@]}"
+}
+
+function check_changed_packages() {
```
Comment (mehmetf) [inline script/check_publish.sh]: It seems that we repeat this function. Perhaps define a common script and include it using `source`?

**Pull request merged**